specification/dartLangSpec.tex

```diff
@@ -11526,27 +11537,30 @@ \subsection{Await Expressions}
 %Otherwise, the value of $a$ is the value of $e$. If evaluation of $e$ raises an exception $x$, $a$ raises $x$.
 
 \commentary{%
-It is typically a compile-time error
-if the function immediately enclosing $a$ is not declared asynchronous.
-E.g., it can be a syntax error because \AWAIT{} has no special meaning
-in the context of a normal function.
-However, \code{\AWAIT($e$)} can also be a function invocation.%
+An await expression can only occur in a function which is declared
+asynchronous. The \AWAIT{} identifier has has no special meaning
+in the context of a normal function, so occurrences of \AWAIT{}
+in those functions does not introduce an await expression.
+However, \code{\AWAIT ($e$)} can be a valid function invocation in
+non-asynchronous functions.%
 }
 
 \rationale{%
-An await expression has no meaning in a synchronous function.
+An await expression could not meaningfully occur in a synchronous function.
 If such a function were to suspend waiting for a future,
 it would no longer be synchronous.%
 }
 
 \commentary{%
 It is not a compile-time error if the type of $e$ is not
-a subtype of \code{Future}.
+a supertype or subtype of \code{Future}.
 Tools may choose to give a hint in such cases.%
 }
 
 \LMHash{}%
-The static type of $a$ is \flatten{T} (\flatten{} is defined in section \ref{functionExpressions}) where $T$ is the static type of $e$.
+The static type of $a$ is $T$.
+If $a$ has a static context type $R$, then the static context type of
+$e$ is \code{FutureOr<$R$>}.
```

Comment:
I'd prefer to have PR #246 landed, introducing context types in a systematic manner. It seems messy to introduce it one place at a time (though it may be faster ;-).